sdk/go/pulumi/stack_reference.go

```diff
@@ -39,6 +39,24 @@ func (s *StackReference) GetIDOutput(name StringInput) IDOutput {
 	}).(IDOutput)
 }
 
+// GetFloat64Output returns a stack output keyed by the given name as an Float64Output
+func (s *StackReference) GetFloat64Output(name StringInput) Float64Output {
+	return s.GetOutput(name).ApplyT(func(out interface{}) float64 {
+		var res float64
+		if out != nil {
+			res = out.(float64)
```

Comment:
This type conversion will panic if out is say, a `string` instead of `float64`. It is not uncommon for users to try and do something like this so we should catch and return an error rather than crashing the runtime. The applier func accepted by `ApplyT` has a form that can propagate errors just for this purpose:

https://github.com/pulumi/pulumi/blob/d4180d9f616100c959c74013ae5191bb2703bc0f/sdk/go/pulumi/types.go#L380

Reply:
I modified the code to return the error in case of a type mismatch. Please check it.